rust/src/transformations/make_stable_expr/expr_fill_nan/mod.rs

```diff
@@ -33,12 +33,19 @@ where
         return fallible!(MakeTransformation, "expected fill_nan expression");
     };
 
+    // only enforce row-by-row context if the fill expression is not broadcastable
+    let expr_domain = if fill.clone().meta().root_names().len() > 0 {
+        input_domain.as_row_by_row()
+    } else {
+        input_domain.clone()
+    };
```

Comment:
Only because there isn't a test that has been updated right beside this file, I'm wondering if both cases are tested?

Reply:
Added.